arangod/IResearch/IResearchView.cpp

```diff
@@ -1109,7 +1110,28 @@ arangodb::Result IResearchView::appendVelocyPackDetailed(
           }
 
           linkBuilder.close();
-          linksBuilderWrapper->add(collectionName, linkBuilder.slice());
+
+          // need to mask out some fields
+          static const std::function<bool(irs::string_ref const& key)> acceptor = [](
+              irs::string_ref const& key
+          )->bool {
+            return key != arangodb::StaticStrings::IndexType
+                && key != StaticStrings::ViewIdField; // ignored fields
```

Comment:
the issue also mentions the "id" field should be suppressed

Reply:
Right, that's why this PR is labeled a WIP. I am trying to understand why masking out  `arangodb::StaticStrings::IndexId` is a problem. It causes the `dump` test to fail, specifically `testView` in `dump-rocksdb.js` and `dump-mmfiles.js`. I'm guessing there is some point in the restore process where the index ID is used to look up a link, rather than it's other properties. Is there a reason we can't use similar logic to the cluster case, i.e. `ensureIndex`?